lib/bolt/result.rb

```diff
@@ -66,9 +66,23 @@ def self.for_task(target, stdout, stderr, exit_code, task)
                             'details' => { 'exit_code' => exit_code } }
       end
 
+      if value.key?('_error')
+        unless value['_error'].is_a?(Hash) && value['_error'].key?('msg')
+          raise Bolt::Error.new(
```

Comment:
....where does this error go? I guess it would probably be the fallback handler in the executor, which is meant to only handle transport bugs. Should we be inserting this as the `_error` key of the result instead of raising it? Maybe with something like `{"details" => { "original_error" => [the original _error] } }`

Reply:
Ah looks like it's caught in `Bolt::Transport::Base#with_events` which calls `Bolt::Result.from_exception`, so that's not so bad. I do think it makes more sense to include this error in the original result though, so we preserve any _other_ result keys the user might care about.